src/components/DataBrowser.vue

```diff
@@ -290,6 +296,38 @@ export default {
           return '';
       }
     },
+    prepareUpsertFolderProps(props) {
+      const newProps = { ...props };
+      delete newProps.location;
+      delete newProps.postUpload;
+      return newProps;
+    },
+    postUpsert() {
```

Comment:
This grouping of functions, `postUpsert`, `postUpload`, `prepareUpsertFolderProps`, and `prepareUploadProps`, are sending me some tingling feelings.

I completely understand your goal, and I think it's reasonable to look for ways to make GWC easier to use.  It's neat that you can get all this functionality out of a single component (from a consumer perspective).

I think the code quality/maintainability cost is just too high here.  DataBrowser is now coupled with each of the other components in a way that I don't feel is necessary.  I would really prefer we didn't bring this logic into DataBrowser.